integration-tests/utils/integration-utils.js

```diff
@@ -159,8 +159,14 @@ const select = async (page, labelContains, text) => {
   return click(page, `//div[starts-with(@id, "react-select-") and contains(normalize-space(.),"${text}")]`)
 }
 
-const waitForNoSpinners = page => {
-  return page.waitForXPath('//*[@data-icon="loadingSpinner"]', { hidden: true })
+// Used to prevent checking loadingSpinner while page is blank
+const waitUntilDomContentLoaded = page => {
+  return page.waitForSelector('div:not(empty):not([id="root"])')
```

Comment:
This CSS selector should be a good start for most pages but probably not sufficient for edge cases.

Reply:
If I understand this correctly, the purpose of this is to wait for the initial "Loading Terra..." message to go away? If so, the name seems misleading. Based on the name, I would have assumed this had something to do with the [DOMContentLoaded event](https://developer.mozilla.org/en-US/docs/Web/API/Window/DOMContentLoaded_event).

Also, this selector looks like it matches the loading message itself, so wouldn't this function return while the loading message is still visible?
https://github.com/DataBiosphere/terra-ui/blob/2c58e0f146245c119c5cdf4d1732591bad950ab1/public/index.html#L17-L30

Finally, was the selector meant to be `:not(:empty)` vs `:not(empty)`?